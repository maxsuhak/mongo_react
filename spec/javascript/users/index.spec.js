import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { HashRouter } from 'react-router-dom'
import New from '../../../app/javascript/src/components/Users/new'

const mockStore = configureStore()

describe('New', () => {
  const store = mockStore({
    table: {
      items: []
    }
  })

  const component = render(
    <Provider store={store}>
      <HashRouter>
        <New />
      </HashRouter>
    </Provider>
  )

  it('matches snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
