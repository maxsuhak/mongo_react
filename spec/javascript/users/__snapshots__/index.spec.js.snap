// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New matches snapshot 1`] = `
Array [
  <h1>
    New User
  </h1>,
  <form
    class="form-horizontal"
    id="create"
  >
    <div
      class="form-group"
    >
      <label
        class="control-label col-sm-3 col-md-2"
        for="first_name"
      >
        First Name
      </label>
      <div
        class="col-sm-3"
      >
        <input
          class="form-control"
          id="first_name"
          required=""
          type="text"
        />
      </div>
    </div>
    <div
      class="form-group"
    >
      <label
        class="control-label col-sm-3 col-md-2"
        for="last_name"
      >
        Last Name
      </label>
      <div
        class="col-sm-3"
      >
        <input
          class="form-control"
          id="last_name"
          required=""
          type="text"
        />
      </div>
    </div>
    <div
      class="form-group"
    >
      <label
        class="control-label col-sm-3 col-md-2"
        for="email"
      >
        Email
      </label>
      <div
        class="col-sm-3"
      >
        <input
          class="form-control"
          id="email"
          required=""
          type="text"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-4"
      >
        <button
          class="btn btn-primary btn-w-m m-r-sm"
          type="submit"
        >
          Create
        </button>
      </div>
      <div
        class="col-sm-4"
      >
        <a
          class="btn btn-primary btn-w-m m-r-sm"
          href="#/users"
        >
          Back
        </a>
      </div>
      <div
        class="col-sm-4"
      />
    </div>
  </form>,
]
`;
